DisplayCAL/madvr.py

```diff
@@ -1487,26 +1487,34 @@ def _wait_for_client(self, addr=None, timeout=1):
                 for c_addr in c_addrs:
                     client = clients.get(c_addr)
                     conn = self._client_sockets.get(c_addr)
-                    if (
-                        client
-                        and client["component"] == "madTPG"
-                        and client.get("confirmed")
-                        and conn
-                        and self._send(conn, "StartTestPattern")
-                    ):
-                        self._client_socket = conn
-                        return True
+                    if client:
```

Comment:
please avoid burrying down this many if statements

Reply:
the reason for this chain of conditionals is to get the precise logging for the reason, without this it just silently ignores responses so working out what broke is one for the debugger only (which is annoying given the thread model and constant flow of messages)

Reply:
Hey @3ll3d00d its clear want you want to achive but you can get it much easier and without raising the complexity by inverting the statements. If something is not, log it.

Reply:
happy to PR something but not entirely sure what the preference is, e.g. I could change to something like

```
                        has_sent_start = None
                        if (
                            client["component"] == b"madTPG"
                            and client.get("confirmed")
                            and conn
                        ):
                            has_sent_start = self._send(conn, "StartTestPattern")
                        safe_print(
                            f"client: {client['component']}, confirmed: {client.get('confirmed')}, connection: {conn}, sent_start: {has_sent_start}"
                        )
                        if has_sent_start:
                            return True
```

which still has all the info required in log but is somewhat simpler

Reply:
I don't think you have to change it right now, its already merged. And its not about what you print but about your nested if statements, that is not considered best practice. "Flat is better than nested." https://peps.python.org/pep-0020/

you could for example also do something like:
```
....
if not client:
    safe_print(f'Ignoring client component : {component_}')
    return False
component_ = client['component']
if component_ != b'madTPG' and not client.get('confirmed'):
    safe_print('madTPG is unconfirmed')
    return False
....
```
Your function or method would have a straight path until it returns (in this case) True at the end. On its way it only handles the exceptions and exits early with the error message.
That way you will have one indent level and in general it is better readable for humans.